tests/cancun/eip1153_tstore/test_tstorage_clear_after_tx.py

```diff
@@ -0,0 +1,118 @@
+"""
+Ethereum Transient Storage EIP Tests
+https://eips.ethereum.org/EIPS/eip-1153
+"""
+
+from typing import Optional
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Bytecode,
+    Environment,
+    EVMCodeType,
+    Initcode,
+    Transaction,
+)
+from ethereum_test_tools.eof.v1 import Container
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import ref_spec_1153
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_1153.git_path
+REFERENCE_SPEC_VERSION = ref_spec_1153.version
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.with_all_evm_code_types
+def test_tstore_clear_after_deployment_tx(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    evm_code_type: EVMCodeType,
+):
+    """
+    This test first creates a contract, which TSTOREs a value 1 in slot 1.
+    After creating the contract, a new tx will call this contract, storing TLOAD(1) into slot 1.
+    The transient storage should be cleared after creating the contract (at tx-level), so
+    the storage should stay empty.
+    """
+    env = Environment()
+
+    init_code = Op.TSTORE(1, 1)
+    deploy_code = Op.SSTORE(1, Op.TLOAD(1))
+
+    code: Optional[Container | Initcode] = None
+    if evm_code_type == EVMCodeType.EOF_V1:
+        code = Container.Init(
+            deploy_container=Container.Code(deploy_code), initcode_prefix=init_code
+        )
+    else:
+        code = Initcode(deploy_code=deploy_code, initcode_prefix=init_code)
+
+    sender = pre.fund_eoa()
+
+    deployment_tx = Transaction(
+        gas_limit=100000,
+        data=code,
+        to=None,
+        sender=sender,
+    )
+
+    address = deployment_tx.created_contract
+
+    invoke_contract_tx = Transaction(gas_limit=100000, to=address, sender=sender)
+
+    txs = [deployment_tx, invoke_contract_tx]
+
+    post = {
+        address: Account(storage={0x01: 0x00}),
+    }
+
+    blockchain_test(genesis_environment=env, pre=pre, post=post, blocks=[Block(txs=txs)])
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.with_all_evm_code_types
+def test_tstore_clear_after_tx(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    evm_code_type: EVMCodeType,
+):
+    """
+    This test first SSTOREs the TLOAD value of key 1 in slot 1. Then, it TSTOREs 1 in slot 1.
+    The second tx will re-call the contract. The storage should stay empty,
+    because the transient storage is cleared after the transaction.
+    """
+    env = Environment()
+
+    runtime_code = Op.SSTORE(1, Op.TLOAD(1)) + Op.TSTORE(1, 1)
+
+    code: Optional[Container | Bytecode] = None
+    if evm_code_type == EVMCodeType.EOF_V1:
+        code = Container.Code(runtime_code)
+    else:
+        code = runtime_code
+
+    account = pre.deploy_contract(code)
```

Comment:
```suggestion
    code = Op.SSTORE(1, Op.TLOAD(1)) + Op.TSTORE(1, 1) + Op.STOP
    account = pre.deploy_contract(code)
```

The logic here shouldn't be necessary for this case, pre.deploy_contract changes its behavior when we use pytest.mark.with_all_evm_code_types to read fixture evm_code_type and automatically wrap the deployed contract within an EOF container (it also should automatically add the STOP).

Might need to remove Bytecode import after this change.

Reply:
Ok, so I can remove `Op.STOP` and then just remove the `evm_code_type` switch logic?

I think I actually copied this logic from another test, so this logic (not automatically switching to EOF) is somewhere else (unless that is necessary for that test). I actually took this from the `test_code_set_txs.py` (EIP 7702), but looking there again I think that logic needs to be there because of the contract creations? :thinking: 

Would the logic still be necessary in `test_tstore_clear_after_deployment_tx` since that deploys a contract?

Reply:
> Ok, so I can remove `Op.STOP` and then just remove the `evm_code_type` switch logic?

Yes, `Op.STOP` is optional because `pre.deploy_contract` detects that the code is not `Container`, it's `Bytecode`, so before automatically wrapping, it checks for the `terminating` flag of the bytecode, and if it's false, it appends the `Op.STOP`.

> I think I actually copied this logic from another test, so this logic (not automatically switching to EOF) is somewhere else (unless that is necessary for that test). I actually took this from the `test_code_set_txs.py` (EIP 7702), but looking there again I think that logic needs to be there because of the contract creations? 🤔
> 
> Would the logic still be necessary in `test_tstore_clear_after_deployment_tx` since that deploys a contract?

Yes I think that's the reason,  if we are dealing with initcode, it needs the switch because we don't handle the `evm_code_type` flag automatically inside the `Initcode` class as we do with `pre.deploy_contract`.

Reply:
Have removed the EOF code switch logic in this specific test.

Reply:
I will merge once the CI passes :)

Reply:
Side-question, why does this PR stay approved even since I just pushed a commit? Shouldn't this require a re-review? I could now put arbitrary code here and merge it to `main`? :thinking:

Reply:
Oh wait, nvm, I cannot merge it myself